lib/interceptor/dump.js

```diff
@@ -0,0 +1,132 @@
+'use strict'
+
+const util = require('../core/util')
+const { InvalidArgumentError, RequestAbortedError } = require('../core/errors')
+const DecoratorHandler = require('../handler/decorator-handler')
+
+class DumpHandler extends DecoratorHandler {
+  #maxSize = 1024 * 1024
+  #abort = null
+  #dumpOnAbort = true
+  #dumped = false
+  #aborted = false
+  #size = 0
+  #reason = null
+  #handler = null
+
+  constructor ({ maxSize, dumpOnAbort }, handler) {
+    super(handler)
+
+    if (maxSize != null && (!Number.isFinite(maxSize) || maxSize < 1)) {
+      throw new InvalidArgumentError('maxSize must be a number greater than 0')
+    }
+
+    if (dumpOnAbort != null && typeof dumpOnAbort !== 'boolean') {
+      throw new InvalidArgumentError('dumpOnAbort must be a boolean')
+    }
+
+    this.#maxSize = maxSize ?? this.#maxSize
+    this.#dumpOnAbort = dumpOnAbort ?? this.#dumpOnAbort
```

Comment:
How about not having this as an option and just have it always enabled? We might also want to have a dump timeout in addition to a maxSize?

Reply:
I believe this goes in line with your suggestions for the PR.
Don't have strong opinion on this one, I believe there might be situations where you don't want to automatically dump on abort, but if we add it, it might never get used. 

So let's simplify and follow your suggestions, if we see use cases or issues we can always add it 👍

Reply:
Regarding the timeout, shouldn't that already be covered by the `bodyTimeout`?
It might look redundant, and to have effect it might require to always set it lower to the `bodyTimeout` set for the Client/Agent